src/accountsdb/snapshots.zig

```diff
@@ -923,6 +1078,171 @@ pub const BankFields = struct {
     }
 };
 
+/// Analogous to https://github.com/anza-xyz/agave/blob/8d1ef48c785a5d9ee5c0df71dc520ee1a49d8168/runtime/src/serde_snapshot.rs#L396
+pub const ExtraFields = struct {
+    lamports_per_signature: u64,
+    snapshot_persistence: ?BankIncrementalSnapshotPersistence,
+    epoch_accounts_hash: ?Hash,
+    versioned_epoch_stakes: VersionedEpochStakesMap,
+    accounts_lt_hash: ?AccountsLtHash,
+
+    pub const INIT_EOF: ExtraFields = .{
+        .lamports_per_signature = 0,
+        .snapshot_persistence = null,
+        .epoch_accounts_hash = null,
+        .versioned_epoch_stakes = .{},
+        .accounts_lt_hash = null,
+    };
+
+    pub fn deinit(extra: *const ExtraFields, allocator: std.mem.Allocator) void {
+        var versioned_epoch_stakes = extra.versioned_epoch_stakes;
+        for (versioned_epoch_stakes.values()) |ves| ves.deinit(allocator);
+        versioned_epoch_stakes.deinit(allocator);
+    }
+
+    pub const VersionedEpochStakesMap = std.AutoArrayHashMapUnmanaged(u64, VersionedEpochStake);
+
+    /// TODO: Investigate this `LtHash` business:
```

Comment:
just a note we have context here: https://github.com/orgs/Syndica/projects/2/views/10?pane=issue&itemId=85238686

Reply:
a60c1c1da8bfca250ac09a60dcc8a9752e8936bc